types/src/system/mint/mod.rs

```diff
@@ -31,13 +31,11 @@ pub trait Mint: RuntimeProvider + StorageProvider + SystemProvider {
         self.write_balance(purse_uref, initial_balance)?;
 
         if !is_empty_purse {
-            // get total supply uref if exists, otherwise create it.
+            // get total supply uref if exists, otherwise error
             let total_supply_uref = match self.get_key(TOTAL_SUPPLY_KEY) {
                 None => {
-                    // create total_supply value and track in mint context
-                    let uref: URef = self.new_uref(U512::zero())?;
-                    self.put_key(TOTAL_SUPPLY_KEY, uref.into())?;
-                    uref
+                    // total supply URef should exist due to genesis
+                    return Err(Error::TotalSupplyNotFound);
```

Comment:
Is this change related to the ticket? It's changing the logic that already exists.

Reply:
Not really, but I think its a valid change to the logic. We shouldn't be creating the URef again anywhere else cause we know it should be present due to genesis, this change prevents that any value `TOTAL_SUPPLY` named key to modified unintientionally